### PR TITLE
[PyT] Update THD sink attention logic for cudnn >=9.18.0

### PR DESCRIPTION
**Pull request by cuichenx**
# Description

THD Sink attention is supported in 9.18.0

GPT-OSS training with thd attention in Megatron-Bridge
image

## Type of change

- [ ] Documentation change (change only to the documentation, either a fix or a new content)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Infra/Build change
- [ ] Code refactoring

## Changes

Please list the changes introduced in this PR:

- Change A
- Change B

# Checklist:

- [ ] I have read and followed the [contributing guidelines](https://github.com/NVIDIA/TransformerEngine/blob/main/CONTRIBUTING.rst)
- [ ] The functionality is complete
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: transformer_engine/pytorch/attention/dot_product_attention/utils.py [hunk 1]
```diff
@@ -716,15 +716,12 @@ def _is_fa3_supported(num_heads, num_gqa_groups, head_dim_qk, head_dim_v, qkv_dt
...
-            )
-            use_fused_attention = False
-            logger.debug(
-                "Disabling UnfusedDotProductAttention for softmax_type = %s and qkv_format = thd",
-                softmax_type,
-            )
-            use_unfused_attention = False
+            if cudnn_version < (9, 18, 0):
+                logger.debug(
+                    "Disabling FusedAttention for softmax_type = %s and qkv_format = thd",
```
Comment (cyanguwa) [inline transformer_engine/pytorch/attention/dot_product_attention/utils.py]: Maybe add the version info in the message? "Disabling FusedAttention for softmax_type = %s, qkv_format = thd and cuDNN version < 9.18"

Comment (cyanguwa) [reply transformer_engine/pytorch/attention/dot_product_attention/utils.py]: We should probably remove L726-L731 below. UnfusedDPA should be disabled for CP for all softmax types, not just non-vanilla ones. We already have that check in L753-L757.

Comment (cuichenx) [reply transformer_engine/pytorch/attention/dot_product_attention/utils.py]: makes sense, addressed these in the latest commit

**Comment by TGS123** [comment]
I'd like to ask a question regarding kernel selection:Why does the forward pass call the cuDNN SM90 interface when the learnable_sink option is enabled with qkv_format="bshd", but reverts to the cuDNN SM80 interface when qkv_format="thd" is selected?

**Comment by cuichenx** [comment]
> I'd like to ask a question regarding kernel selection:Why does the forward pass call the cuDNN SM90 interface when the learnable_sink option is enabled with qkv_format="bshd", but reverts to the cuDNN SM80 interface when qkv_format="thd" is selected?

Hi @TGS123, you're right this was a bug in the cudnn 9.18 release. It will be fixed in 9.19!

**Comment by TGS123** [comment]
您好！您的邮件我已收到，谢谢啦。
